lib/Onyx.js

```diff
@@ -780,6 +780,39 @@ function mergeCollection(collectionKey, collection) {
         });
 }
 
+/**
+ * Insert API responses and lifecycle data into Onyx
+ *
+ * @param {Object} data An object containing the Onyx data to set
+ * @param {String} data.onyxMethod One of [set, merge] Onyx methods
+ * @param {String} data.key ONYXKEY to set
+ * @param {*} data.value Value to store in ONYXKEY
+ */
+function update(data) {
+    // First, validate the Onyx object is in the format we expect
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (!_.contains(['set', 'merge'], onyxMethod)) {
+            throw new Error(`Invalid onyxMethod ${onyxMethod} in Onyx update.`);
+        }
+        if (!_.isString(key)) {
+            throw new Error(`Invalid ${typeof key} key provided in Onyx update. Onyx key must be of type string.`);
+        }
+        if (typeof value === 'undefined') {
```

Comment:
Use `_.isUndefined()` to be consistent with other underscore methods.

Reply:
Also, I think it's OK for `undefined` to be passed with `set()` is that right? It would be expected to remove the Onyx key entirely.

Reply:
That seems right based on [this comment](https://github.com/Expensify/react-native-onyx/blob/8772195796b9a0c9c6fec5a3feeeb214f2b6a5ca/lib/Onyx.js#L631). However, [this line](https://github.com/Expensify/react-native-onyx/blob/8772195796b9a0c9c6fec5a3feeeb214f2b6a5ca/lib/Onyx.js#L634) seems to only check for `null` values. Maybe I'm missing something though.

Reply:
I'll remove this check since it seems like we want to support `undefined` values as well.

Reply:
Hm, I don't think that's quite right. That line is for merging, and it just means that if a `null` value is passed to `merge()` then it will get ignored (hence, it is "deleted" from the data being merged into Onyx).

The [`setItem()` method in LocalForage](https://localforage.github.io/localForage/#data-api-setitem) doesn't seem to mention what happens if you pass `undefined` as the value, but it might be worth experimenting with to verify the expected behavior!

Reply:
Hmm that makes sense. I experimented a bit and it seems that passing undefined does remove the key. I'm removing that check then. Thanks!